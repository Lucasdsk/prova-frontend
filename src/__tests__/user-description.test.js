import React from 'react'
import UserDescription from './../components/user-description'

describe("<UserDescription>", () => {
  it("Render UserDescription", () => {
    const UserDescriptionComponent = shallow(<UserDescription />)
    expect(UserDescriptionComponent).toMatchSnapshot()
  })

  it("Should have a name", () => {
    const UserDescriptionComponent = shallow(<UserDescription name="Usuario" />)
    expect(UserDescriptionComponent.find('.user-description__name').text()).toBe('Usuario')
  })

  it("Should have options when the props 'showOptions', 'options' and the state 'optionsVisible' is 'true'", () => {
    const CustomOptions = () => (<div className="customOptions">Options</div>)
    const UserDescriptionComponent = shallow(<UserDescription
      name="Usuario"
      showOptions
      options={<CustomOptions />}
    />)
    
    expect(UserDescriptionComponent.find('.user-description__options-wrapper').length).toBe(1)
    UserDescriptionComponent.simulate('click')
    expect(UserDescriptionComponent.state().optionsVisible).toBe(true)
    expect(UserDescriptionComponent.find('.user-description__options-container').length).toBe(1)
    expect(UserDescriptionComponent.find('.user-description__options').length).toBe(1)
    expect(UserDescriptionComponent.contains(<CustomOptions/>)).toBe(true)
  })
})
