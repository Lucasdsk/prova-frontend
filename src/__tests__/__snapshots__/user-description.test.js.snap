// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UserDescription> Render UserDescription 1`] = `
<div
  className="user-description"
>
  <div
    className="user-description__img"
    style={
      Object {
        "backgroundImage": "url(test-file-stub)",
        "height": 50,
        "width": 50,
      }
    }
  />
  <span
    className="user-description__name"
  />
</div>
`;
